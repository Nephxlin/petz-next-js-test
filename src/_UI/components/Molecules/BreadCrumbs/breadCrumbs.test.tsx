import { render, screen } from '@testing-library/react';
import Breadcrumbs from './index';
import mockRouter from 'next-router-mock';

jest.mock('next/router', () => jest.requireActual('next-router-mock'))

describe('Breadcrumbs component', () => {
  test('renders Breadcrumbs component with provided props', () => {

    mockRouter.push("/quem-somos");

    const { container } = render(<Breadcrumbs />);
    expect(container).toMatchSnapshot()

    const breadcrumbsList = screen.getByTestId('breadcrumbs-list');
    expect(breadcrumbsList).toBeInTheDocument();

    const homeLink = screen.getByText('Home');
    expect(homeLink).toBeInTheDocument();

    const currentLink = screen.getByTestId('breadcrumbs-items')
    expect(currentLink).toHaveTextContent('Quem somos')
  });
});
