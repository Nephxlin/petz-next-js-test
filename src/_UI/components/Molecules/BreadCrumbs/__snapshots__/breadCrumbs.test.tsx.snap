// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component renders Breadcrumbs component with provided props 1`] = `
.c0 {
  display: flex;
  align-items: center;
  list-style: none;
  gap: 4px;
}

.c1 {
  display: flex;
  align-items: center;
  font-size: 12px;
  font-weight: 700;
  color: #ffffff;
  gap: 4px;
  text-decoration: none;
}

.c1:hover {
  text-decoration: underline;
}

.c2 {
  display: flex;
  align-items: center;
  font-size: 12px;
  font-weight: 700;
  color: #ffffff;
  gap: 4px;
  text-decoration: none;
}

.c2:before {
  content: ' > ';
  font-weight: 300;
}

<div>
  <ul
    class="c0"
    data-testid="breadcrumbs-list"
  >
    <a
      href="/"
    >
      <li
        class="c1"
      >
        Home
      </li>
    </a>
    <li
      class="c2"
      data-testid="breadcrumbs-items"
    >
      Quem somos
    </li>
  </ul>
</div>
`;
